### PR TITLE
Xrddev 519 add local group

### PR DESCRIPTION
**Pull request by Riippi**
## Description
- User can create a new local group
- Fixes for unit tests
- Fix three new npm audit warnings
- Fix groups apis

Links:
https://jira.niis.org/browse/XRDDEV-519
https://gofore.invisionapp.com/share/ZUQYEC87P5X#/screens/352143364
https://jenkins.niis.org/view/api-based-ui/job/rest-ui-build-pull-request/41/console

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have followed the agreed [Version Control Practices](https://confluence.niis.org/pages/viewpage.action?spaceKey=XRDDEV&title=Version+Control+Practices)
- [x] My changes generate no new warnings or errors (e.g. Javascript console, Java stdout)
- [ ] I have made corresponding changes to the documentation
- [ ] The new code has sufficient test coverage
- [x] The build, unit and integration tests pass
- [ ] There is a link to a successful Jenkins build
- [x] No new npm audit issues, or new issues have been added to [Front-end build process](https://confluence.niis.org/display/XRDDEV/Front-end+build+process) tracking table and accepted
- [ ] New backlog items that have been created (such as "not implementing this acceptance criteria now") are mentioned + linked in Jira task comments
- [ ] All task outputs (PR, documentation, UI design, etc) is listed in a Jira comment so that it is easy for reviewer to check
- [ ] Deviations from acceptance criteria listed in comments (also if "this criteria was removed")

### CODE REVIEW

File: src/proxy-ui-api/frontend/src/views/LocalGroup.vue [hunk 1]
```diff
@@ -242,11 +245,20 @@ export default Vue.extend({
     },
     doDeleteGroup(): void {
       this.confirmGroup = false;
-      // TODO placeholder. will be done in future task
+
+      axios
+        .delete(`/groups/${this.groupId}`)
+        .then((res) => {
```
Comment (carohauta) [inline src/proxy-ui-api/frontend/src/views/LocalGroup.vue]: Unused parameter `res`

File: src/proxy-ui-api/frontend/src/components/NewGroupDialog.vue [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+      this.clearForm();
+      this.$emit('cancel');
+    },
+    save(): void {
+      axios
+        .post(`/clients/${this.id}/groups`, {
+          code: this.code,
+          description: this.description,
+        })
+        .then((res) => {
```
Comment (carohauta) [inline src/proxy-ui-api/frontend/src/components/NewGroupDialog.vue]: Unused parameter `res`

File: src/proxy-ui-api/frontend/src/views/LocalGroup.vue (hunk 1 above)
Comment (Riippi) [reply src/proxy-ui-api/frontend/src/views/LocalGroup.vue]: Fixed

File: src/proxy-ui-api/frontend/src/components/NewGroupDialog.vue (hunk 2 above)
Comment (Riippi) [reply src/proxy-ui-api/frontend/src/components/NewGroupDialog.vue]: Fixed

**Pull request merged**